### PR TITLE
6. Add event log subscriptions.

### PR DESCRIPTION
**Pull request by nikhitashanker77**
- Add a utility class windows_events/subscriptions.go that creates subscriptions.
- Add the creation of subscriptions to the workflow scheduler.

Tests will be added in future pr.

### CODE REVIEW

**Review by nikhitashanker77 — COMMENTED** [review]
1. github.com/ofcoursedude/gowinlog
      error: The parameter is incorrect => unable to solve

2. windows-events/system_windows.go at master · LiamHaworth/windows-events (github.com) => Example works
		○ Problem: callback function takes only 1 parameter which is the event but we want callback to also take the workflow + other info as a parameter
		○ We also need to do error handling ourselves.
		EventData can only hold string-string key value pairs.
		
3. I have now successfully created a helloworld program + implementation in the PR by adapting windows-events/system_windows.go at master · LiamHaworth/windows-events (github.com)  
	- Do I need to document anything if I adapted some code?
	- Helloworld program demonstrates that subscribing and running workflows is event driven.

		
My implementation assumes each node has a wevtapi.dll file

File: windows_events/subscriber.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
+package windows_events
+
+import "C"
```
Comment (NoumbissiValere) [inline windows_events/subscriber.go]: @nikhitashanker77 do we know why are importing C here? I don't see any usage for it in the code.

File: main.go [hunk 2]
```diff
@@ -18,14 +19,13 @@ type Main struct {
...
 
-	var workflow workflows.InterfaceWorkflow = mainObj.workflowFactory.constructWorkflow("dummy_workflow")
-	var workflowContext *nodes.WorkflowContext = nodes.NewWorkflowContext()
-	workflow.Run(workflow, workflowContext)
-
-	for index, node := range workflow.GetNodes() {
-		mainObj.logger.Println(fmt.Sprintf("Result for node index %d=%s", index, node.GetResult(node, workflowContext).(string)))
-	}
+	// Setup main such that main does not exit unless there is a keyboard interrupt.
+	go forever()
```
Comment (NoumbissiValere) [inline main.go]: Even without this, main will not exit because of the channel we created unless it is interrupted.  so we can remove this forever method.

**Review by NoumbissiValere — CHANGES_REQUESTED** [review]
Hi @nikhitashanker77 , Please look at the minor changes requested

File: windows_events/subscriber.go (hunk 1 above)
Comment (nikhitashanker77) [reply windows_events/subscriber.go]: Yes, I don't need it here I have removed it now.

File: main.go (hunk 2 above)
Comment (nikhitashanker77) [reply main.go]: I see, I have removed the forever method.

**Review by NoumbissiValere — APPROVED** [review]
Looks good to me. Leaving for Varun or Sindhu and Deepti to validate the functionalities

**Comment by nikhitashanker77** [comment]
Set GOMAXPROCS to make code single-threaded.

https://pkg.go.dev/runtime

File: workflow_scheduler.go [hunk 3]
```diff
@@ -56,7 +66,7 @@ func newWorkflowScheduler(schedulesJson []byte, workflowFactory WorkflowFactory)
 				panic("Given schedule type not supported.")
```
Comment (varunkrishna26) [inline workflow_scheduler.go]: This shouldn't be a panic really. We should just log and continue with the remaining schedules. Something to fix in a future PR.

**Review by varunkrishna26 — APPROVED** [review]

File: workflow_scheduler.go (hunk 3 above)
Comment (nikhitashanker77) [reply workflow_scheduler.go]: I see, fixed in current PR.

**Pull request merged**